docs/docs/contribute.md

```diff
@@ -1 +1,194 @@
+to never work on the main branch, as this allows you to easily get back to a
+working state of the code if needed (e.g., if you’re working on multiple
+changes at once, or need to pull in recent changes from someone else to get
+your new feature to work properly). In most cases you should make PRs into the
+upstream’s main branch.
+
+**Step 10**: Develop your code on your feature branch on the computer, using
+Git to do the version control. When you’re done editing, add changed files
+using `git add` and then `git commit` to record your changes:
+```
+git add modified_files
+git commit -m "description of your commit"
+```
+Then push the changes to your GitHub account with:
+```
+git push -u origin my_feature
+```
+The `-u` flag ensures that your local branch will be automatically linked with the remote branch, so you can later use `git push` and `git pull` without any extra arguments.
+
+**Step 11**: Follow [these](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) instructions to create a pull request from your fork.
+This will send a notification to `sbi` maintainers and trigger reviews and comments
+regarding your contribution.
+
+!!! note
+    It is often helpful to keep your local feature branch synchronized
+    with the latest changes of the main `sbi` repository:
+    ```
+    git fetch upstream
+    git merge upstream/main
+    ```
+
+### Style conventions and testing
+
+All our docstrings and comments are written following the [Google
+Style](http://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).
+
+For code linting and formating, we use [`ruff`](https://docs.astral.sh/ruff/),
+which is installed alongside `sbi`.
+
+You can exclude slow tests and those which require a GPU with `pytest -m "not slow and not gpu"`.
+Additionally, we recommend to run tests with `pytest -n auto -m "not slow and not gpu"` in parallel.
+GPU tests should probably not be run this way.
+If you see unexpected behavior (tests fail if they shouldn't), try to run them without `-n auto` and see if it persists.
+When writing new tests and debugging things, it may make sense to also run them without `-n auto`.
+
+When you create a PR onto `main`, our Continuous Integration (CI) actions on
+GitHub will perform the following checks:
+
+- **`ruff`** for linting and formatting (including `black`, `isort`, and `flake8`)
+- **[`pyright`](https://github.com/Microsoft/pyright)** for static type checking.
+- **`pytest`** for running a subset of fast tests from our test suite.
+
+If any of these fail, try reproducing and solving the error locally:
+
+- **`ruff`**: Make sure you have `pre-commit` installed locally with the same version as specified in the [requirements](pyproject.toml). Execute it
+ using `pre-commit run --all-files`. `ruff` tends to give informative error
+  messages that help you fix the problem.
```

Comment:
One thing we should add here: 
`Note that `pre-commit` only detects problems with `ruff` linting and formatting, but does not fix them. You can fix them either by running `ruff check . --fix` (linting), followed by `ruff format . --fix` (formatting), or by hand.